extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeploy.java

```diff
@@ -32,25 +31,28 @@ public boolean getAsBoolean() {
         if (serverFound) {
             return true;
         }
-        try (final KubernetesClient client = KubernetesClientUtils.createClient()) {
+
+        KubernetesClient client = KubernetesClientUtils.createClient();
+        String masterURL = client.getConfiguration().getMasterUrl();
+        try {
+            masterURL = client.getConfiguration().getMasterUrl();
             //Let's check id we can connect.
             RootPaths paths = client.rootPaths();
-            log.info("Found kubernetes server.");
+            log.info("Kubernetes API Server at '" + masterURL + "' successfully contacted.");
             serverFound = true;
+            client.close();
             return true;
         } catch (Exception e) {
-            if (!alreadyWarned) {
-                if (e.getCause() instanceof SSLHandshakeException) {
-                    String message = "Although a Kubernetes deployment was requested, it will however not take place because the API Server certificates are not trusted. The certificates can be configured using the relevant configuration propertiers under the 'quarkus.kubernetes-client' config root, or \"quarkus.kubernetes-client.trust-certs=true\" can be set to explicitly trust the certificates (not recommended)";
-                    log.warn(message);
-                } else {
-                    log.error(
-                            "Although a Kubernetes deployment was requested, it will however not take place because there was an error during communication with the API Server: "
-                                    + e.getMessage());
-                }
-                alreadyWarned = true;
+            if (e.getCause() instanceof SSLHandshakeException) {
+                throw new RuntimeException(
+                        "Although a Kubernetes deployment was requested, it however cannot take place because the API Server certificates are not trusted. The certificates can be configured using the relevant configuration propertiers under the 'quarkus.kubernetes-client' config root, or \"quarkus.kubernetes-client.trust-certs=true\" can be set to explicitly trust the certificates (not recommended)",
```

Comment:
should this not also include the masterURL?

Reply:
Ah yeah, I missed it